python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1573 @@
+                if key in ["CENT_X", "CENT_Y"]:
+                    row[f"{prefix}_{col_suffix}"] = val * u.pixel
+                else:
+                    row[f"{prefix}_{col_suffix}"] = val
+                # field_x/y are aliases for fx/fy
+                if key == "FX":
+                    row[f"{prefix}_field_x"] = val * u.deg
+                elif key == "FY":
+                    row[f"{prefix}_field_y"] = val * u.deg
+            else:
+                # Add appropriate units for NaN values
+                if key in ["CENT_X", "CENT_Y"]:
+                    row[f"{prefix}_{col_suffix}"] = np.nan * u.pixel
+                else:
+                    row[f"{prefix}_{col_suffix}"] = np.nan
+                if key == "FX":
+                    row[f"{prefix}_field_x"] = np.nan * u.deg
+                elif key == "FY":
+                    row[f"{prefix}_field_y"] = np.nan * u.deg
+
+    def validate(self) -> None:
+        """Validate configuration parameters."""
+        # Validate defocal distances
+        if abs(self.config.intraDfcDist) < self.MIN_DEFOCAL_DISTANCE_MM:
+            raise ValueError(f"intraDfcDist too small: {self.config.intraDfcDist} mm")
+        if abs(self.config.extraDfcDist) < self.MIN_DEFOCAL_DISTANCE_MM:
+            raise ValueError(f"extraDfcDist too small: {self.config.extraDfcDist} mm")
+        if self.config.intraDfcDist >= 0:
+            self.log.warning("intraDfcDist should typically be negative (inward defocus)")
+        if self.config.extraDfcDist <= 0:
+            self.log.warning("extraDfcDist should typically be positive (outward defocus)")
+
+        # Validate crop size
+        if self.config.cropSize not in self.STANDARD_CROP_SIZES:
+            self.log.warning("Non-standard cropSize: %d", self.config.cropSize)
+
+        # Validate Noll indices
+        if len(self.config.nollIndices) == 0:
+            raise ValueError("nollIndices cannot be empty")
+        if any(idx < 1 for idx in self.config.nollIndices):
+            raise ValueError("Noll indices must be >= 1")
+
+    def _isPytorchTensor(self, obj: Any) -> bool:
+        """Check if object is a PyTorch tensor."""
+        try:
+            # Use getattr to avoid hasattr checks
+            cpuMethod = getattr(obj, "cpu", None)
+            numpyMethod = getattr(obj, "numpy", None)
+            return cpuMethod is not None and numpyMethod is not None
+        except Exception:
+            return False
+
+    def _toNumpy(self, obj: Any) -> np.ndarray:
+        """Convert PyTorch tensor or other object to numpy array."""
+        if self._isPytorchTensor(obj):
+            return obj.cpu().numpy()
+        elif isinstance(obj, np.ndarray):
+            return obj
+        else:
+            return np.array(obj)
+
+    def _getMetadataArray(self, metadata: PropertyList, key: str, numStamps: int, default: Any = 0.0) -> list:
+        """Extract metadata array safely with default fallback.
```

Comment:
Clarify this is the metadata for donutStamps since there are multiple types of metadata involved in this task.